src/base/bittorrent/torrentimpl.cpp

```diff
@@ -180,20 +187,43 @@ namespace
             }
             trackerEntry.endpoints.append(trackerEndpoint);
 
-            trackerEntry.numSeeds = std::max(trackerEntry.numSeeds, endpoint.scrape_complete);
-            trackerEntry.numLeeches = std::max(trackerEntry.numLeeches, endpoint.scrape_incomplete);
-            trackerEntry.numDownloaded = std::max(trackerEntry.numDownloaded, endpoint.scrape_downloaded);
+            trackerEntry.numPeers = std::max(trackerEntry.numPeers, trackerEndpoint.numPeers);
+            trackerEntry.numSeeds = std::max(trackerEntry.numSeeds, trackerEndpoint.numSeeds);
+            trackerEntry.numLeeches = std::max(trackerEntry.numLeeches, trackerEndpoint.numLeeches);
+            trackerEntry.numDownloaded = std::max(trackerEntry.numDownloaded, trackerEndpoint.numDownloaded);
         }
 #endif
 
         if (numEndpoints > 0)
         {
             if (numUpdating > 0)
+            {
                 trackerEntry.status = TrackerEntry::Updating;
+            }
             else if (numWorking > 0)
+            {
                 trackerEntry.status = TrackerEntry::Working;
+
+                const auto it = std::find_if(trackerEntry.endpoints.cbegin(), trackerEntry.endpoints.cbegin()
+                             , [](const TrackerEntry::EndpointStats &endpoint)
+                {
+                    return ((endpoint.status == TrackerEntry::Working) && !endpoint.message.isEmpty());
+                });
+                if (Q_UNLIKELY(it != trackerEntry.endpoints.cend()))
+                    trackerEntry.message = it->message;
```

Comment:
I still don't see any reasonable way to summarize messages from multiple "tracker endpoints", so I just pick the first one that fits.
We really should display info for each "tracker endpoint" separately.